rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -225,6 +226,11 @@ struct convert<rosbag2_storage::BagMetadata>
       metadata.files =
         node["files"].as<std::vector<rosbag2_storage::FileInformation>>();
     }
+
+    if (node["custom"]) {
```

Comment:
Out of curiosity, is it really a valid check `if (node["custom"])` ?
Will not it always return true as in case with std::unordered_map?

Reply:
`operator[]` of YAML-cpp Node will not create the element if the key doesn't exist.